src/management/list.ts

```diff
@@ -35,9 +36,71 @@ export async function listFirebaseProjects(
   }
 }
 
+/**
+ * Send recurring API requests to list all Firebase apps of a Firebase project
+ * @return a promise that resolves to the list of all Firebase apps.
+ */
+export async function listFirebaseApps(
+  projectId: string,
+  options: { platform?: AppPlatform; pageSize?: number } = {}
+): Promise<AppMetadata[]> {
+  const apps: AppMetadata[] = [];
+  const { platform, pageSize } = options;
+  try {
+    let nextPageToken = "";
+    do {
+      const response = await getPageApiRequest(
+        getListAppsResourceString(projectId, platform),
+        pageSize || APP_LIST_PAGE_SIZE,
+        nextPageToken
+      );
+      if (response.body.apps) {
+        const appsOnPage = response.body.apps.map(
+          // app.platform does not exist if we use the endpoint for a specific platform
+          (app: any) => (app.platform ? app : { ...app, platform })
+        );
+        apps.push(...appsOnPage);
+      }
+      nextPageToken = response.body.nextPageToken;
+    } while (nextPageToken);
+
+    return apps;
+  } catch (err) {
+    logger.debug(err.message);
+    throw new FirebaseError(
+      `Failed to list Firebase ${platform ? platform + " " : ""}` +
+        "apps. See firebase-debug.log for more info.",
+      {
+        exit: 2,
+        original: err,
+      }
+    );
+  }
+}
+
+function getListAppsResourceString(projectId: string, platform?: AppPlatform): string {
+  let resourceSuffix;
+  switch (platform) {
+    case AppPlatform.IOS:
+      resourceSuffix = "/iosApps";
+      break;
+    case AppPlatform.ANDROID:
+      resourceSuffix = "/androidApps";
+      break;
+    case AppPlatform.WEB:
+      resourceSuffix = "/webApps";
+      break;
+    default:
+      resourceSuffix = ":searchApps";
```

Comment:
What's the use case for this `:searchApps`? Is this for a "search" or "list all" functionality for later?

If so, `AppPlatform` (the `enum`) should include an `ALL_PLATFORMS` value that is used here (and wherever this is scenario originates).

The `default` case, in that scenario, probably should be to throw an error if the `platform` isn't recognized

Reply:
`:searchApps` is "a convenience method that lists all available Apps for the specified FirebaseProject".
https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects/searchApps

Okay, including an `ALL_PLATFORMS` value in `AppPlatform` is fine with me. @TrCaM, let's do that and remove the ability for `AppPlatform` to be null/undefined internally. The user-facing portion of the command itself should remain the same.

Reply:
`:searchApps` will list all apps from a project (I know `searchApps` seems to be a confusing name).

I agree that we should move forward introducing `ALL_PLATFORMS` to the enum, thus we can have more clean code

Reply:
Done